File: backend/domain/beneficiary.py
from dataclasses import dataclass
from datetime import datetime


@dataclass(slots=True)
class Beneficiary:
    user_id: int
    name: str
    account_number: str
    sort_code: str
    bank_name: str
    reference: str | None = None
    is_active: bool = True
    id: int | None = None
    created_at: datetime | None = None
    updated_at: datetime | None = None

File: frontend/src/types/beneficiary.ts
export interface Beneficiary {
  id: number;
  user_id: number;
  name: string;
  account_number: string;
  sort_code: string;
  bank_name: string;
  reference: string | null;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface CreateBeneficiaryRequest {
  name: string;
  account_number: string;
  sort_code: string;
  bank_name: string;
  reference?: string | null;
}

export interface UpdateBeneficiaryRequest {
  name?: string;
  account_number?: string;
  sort_code?: string;
  bank_name?: string;
  reference?: string | null;
  is_active?: boolean;
}
